bumble/device.py

```diff
@@ -1717,6 +1731,9 @@ def on_connection_failure(self, connection_handle, error_code):
             'hci',
             HCI_Constant.error_name(error_code)
         )
+        error.transport = transport
+        error.connection_handle = connection_handle # FIXME: Connection handle sounds to be a dummy value here
```

Comment:
you're right, the `connection_handle` here isn't useful. There's a `connection_handle` in the underlying HCI event, but in the case of a connection failure, it means nothing. So I suggest actually removing this parameter from the `on_connection_failure` method altogether.